### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by katerinagutich**
Created form to add new tasks to the list, added buttons to remove tasks.

### CODE REVIEW

File: src/ToDo/AddTodo.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React, {useState} from 'react'
+import PropTypes from 'prop-types'
+
+function useInputValue(defaultValue = '') {
+    const [value, setValue] = React.useState(defaultValue)
```
Comment (DmitrySannikov) [inline src/ToDo/AddTodo.js:5]: just `useState(defaultValue);` you don't need `React` as you already destructured useState upper in imports

File: src/ToDo/AddTodo.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React, {useState} from 'react'
+import PropTypes from 'prop-types'
+
+function useInputValue(defaultValue = '') {
```
Comment (DmitrySannikov) [inline src/ToDo/AddTodo.js:4]: I think it is too complicated, it is better to move this to the AddTodo component without this return object

File: src/App.js [hunk 3]
```diff
@@ -1,20 +1,53 @@
 import React from 'react'
```
Comment (DmitrySannikov) [inline src/App.js:1]: it is better to remove context from the app and pass all methods (toggleTodo, removeTodo, addTodo) through props

**Review by DmitrySannikov — COMMENTED** [review]
please, in js code add a semicolon where it is possible

File: src/ToDo/ToDoItem.js [hunk 4]
```diff
@@ -1,10 +1,34 @@
...
+                <input
+                    type="checkbox"
+                    checked={item.completed}
+                    onChange={() => onSelect(item.id)}
+                />
+                <strong>{index + 1}</strong>
+                &nbsp;
+                {item.title}
+            </span>
+            <button className="rm" onClick={onDelete.bind(null, item.id)}>&times;</button>
```
Comment (DmitrySannikov) [inline src/ToDo/ToDoItem.js:23]: bind is not the best solution, arrow function used more often
`onClick={() => onDelete(item.id)}`

File: src/ToDo/ToDoItem.js [hunk 5]
```diff
@@ -1,10 +1,34 @@
...
         </li>
-    )
-}
\ No newline at end of file
+    );
+}
+
+ToDoItem.propTypes = {
+    item: PropTypes.object.isRequired,
```
Comment (DmitrySannikov) [inline src/ToDo/ToDoItem.js:29]: PropTypes has a shape method it is better to use it here
https://reactjs.org/docs/typechecking-with-proptypes.html

File: src/context.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import React from 'react'
```
Comment (DmitrySannikov) [inline src/context.js:1]: I think we don't use context anymore

File: src/App.js [hunk 7]
```diff
@@ -1,20 +1,50 @@
...
+    const [todos, setTodos] = React.useState(
+        [
+            {id: 1, completed: false, title: "Buy milk"},
+            {id: 2, completed: false, title: "Buy bread"},
+            {id: 3, completed: false, title: "Buy chocolate"}
+        ]
+    )
+
+    const toggleTodo = id => {
+        let todosWithUpdatedStatus = todos.map(todo => {
```
[the remark this answers is not in the document]
Comment (katerinagutich) [reply src/App.js:16]: I thought if a variable refers to an array that can be changed, then we need to use `let` instead

File: src/ToDo/AddTodo.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+
+function AddTodo({onCreate}) {
+
+    const [value, setValue] = useState('');
+
+    const input = {
```
[the remark this answers is not in the document]
Comment (katerinagutich) [reply src/ToDo/AddTodo.js:8]: Removed the object, Now the code is definitely more readable. Thanks!

File: src/context.js (hunk 6 above)
Comment (katerinagutich) [reply src/context.js:1]: I forgot about the file, will delete it.

**Review by DmitrySannikov — APPROVED** [review]

**Pull request merged**